test/unittests/commands/ingest_xmlupload/test_apply_ingest_id.py

```diff
@@ -0,0 +1,72 @@
+from lxml import etree
+
+from dsp_tools.commands.ingest_xmlupload.apply_ingest_id import replace_filepath_with_sipi_id
+
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+
+class TestReplaceBitstreamPaths:
+    def test_all_good(self) -> None:
+        xml = (
+            "<knora>"
+            '<resource label="Fluffy Image" restype=":Image2D" id="Fluffy1" permissions="res-default">'
+            '<bitstream permissions="prop-default">images/Fluffy.jpg</bitstream>'
+            "</resource>"
+            "</knora>"
+        )
+        root = etree.ElementTree(etree.fromstring(xml))
+        reference_dict = {"images/Fluffy.jpg": "fluffy_id"}
+        res_tree, res_msg = replace_filepath_with_sipi_id(root, reference_dict)
+        assert not res_msg.media_no_id
+        assert not res_msg.unused_media_paths
+        expected = (
+            b"<knora>"
+            b'<resource label="Fluffy Image" restype=":Image2D" id="Fluffy1" permissions="res-default">'
+            b'<bitstream permissions="prop-default">fluffy_id</bitstream>'
+            b"</resource></knora>"
+        )
+        assert etree.tostring(res_tree) == expected
```

Comment:
Me too, I strongly suggest comparing the content of the XML, not the string serialization. I made bad experiences with comparing XML strings...
Here an example how you can do it, and it's even simpler and shorter:

```suggestion
        assert res_tree.getroot()[0][0].text == "fluffy_id"
```